packages/storage/amplify_storage_s3_dart/lib/src/exception/s3_storage_exception.dart

```diff
@@ -211,6 +213,16 @@ class S3Exception extends StorageException {
         recoverySuggestion: _fileIssueMessage,
       );
 
+  /// Creates a [NetworkException] over a [AWSHttpException] thrown from
+  /// [s3.S3Client] APIs.
+  static NetworkException fromAWSHttpException(AWSHttpException exception) {
```

Comment:
other categories do similar, `throw NetworkException on AWSHttpException`. How about to define this at amplify_core (types/exception/network_exception.dart) instead?

Reply:
Agreed, will refactor (actually remove S3Exception) in the following PRs, will move this parser around accordingly as part of the following PR.